build.py

```diff
@@ -127,15 +131,36 @@ def _write_v2(self, outputdir):
 
     def _write_cluster_version_mapping(self, outputdir):
         srcpath = self.sourcedir / "templates_v2" / "cluster_version_mapping.json"
+        self._validate_cluster_version_mapping(srcpath)
         dstpath = outputdir / "cluster_version_mapping.json"
         logger.info(f"Writing cluster_version_mapping.json: {dstpath}")
         # preserve non-standard formatting of the file
         shutil.copy(srcpath, dstpath)
-        self._validate_cluster_version_mapping(dstpath)
 
     def _validate_cluster_version_mapping(self, filepath):
         content = self._load_json(filepath)
         self._assert_json_schema(filepath, content, "cluster_version_mapping.schema.json")
+        self._assert_cluster_version_mapping_order(filepath, content)
+
+    def _assert_cluster_version_mapping_order(self, filepath, content):
+        i = 0
+        v1 = semver.Version.parse(content[0][0])
+        for raw_version, _ in content[1:]:
```

Comment:
You can also use `enumerate` to get the index rather than having `i` as a separate variable.

Reply:
I didn't like enumerate when not starting the loop at index 0. I'm using it now since I refactored the code as discussed in the other thread. Please check again.